pkg/dialect/types.go

```diff
@@ -0,0 +1,77 @@
+package dialect
+
+import (
+	"github.com/golang/protobuf/protoc-gen-go/descriptor"
+	"github.com/gophersbd/ormpb/protobuf"
+)
+
+// Supported Data type
+const (
+	Int       = "INT"
+	BigInt    = "BIGINT"
+	Varchar   = "VARCHAR"
+	Text      = "TEXT"
+	Float     = "FLOAT"
+	Double    = "DOUBLE"
+	Numeric   = "NUMERIC"
+	Bool      = "BOOL"
+	Boolean   = "BOOLEAN"
+	Serial    = "SERIAL"
+	BigSerial = "BIGSERIAL"
+	Integer   = "INTEGER"
+	Timestamp = "TIMESTAMP"
+)
+
+// SQLType for Data type and Data size
+type SQLType struct {
+	Name          string
+	DefaultLength int
+}
+
+// type2SQLType converts Proto type to DB Data type
+func type2SQLType(filedType descriptor.FieldDescriptorProto_Type, typeName string) (st SQLType) {
+	switch filedType {
+	case descriptor.FieldDescriptorProto_TYPE_DOUBLE:
```

Comment:
Make sure we also consider the `type` field options if that is provided?

Reply:
First checked for tag. If not found then called this method